src/app/shared/components/city-autocomplete/city-autocomplete.component.html

```diff
@@ -1,11 +1,12 @@
 <form class="city-search">
   <mat-form-field appearance="none">
-    <input matInput type="text" matInput [formControl]="cityControl" [matAutocomplete]="auto">
+    <input type="text" matInput [formControl]="cityControl" [matAutocomplete]="auto">
     <mat-autocomplete autoActiveFirstOption #auto="matAutocomplete" (onKeyPress)="onSelect($event)"
       (optionSelected)="onSelect($event)" [displayWith]="displayCityName">
       <mat-option [disabled]="isCity | async" *ngFor="let city of cities$ | async" [value]="city">
         {{city.name}}
       </mat-option>
     </mat-autocomplete>
+    <app-city-confirmation *ngIf="isConfirmCity"></app-city-confirmation>
```

Comment:
you can use async pipe here without creating subscription in ts

Reply:
done